DR2-1440 Improved logging.

MancunianSam opened the pull request:
I've added a few lines so we can see the flow through the lambda. It's
not necessary to add log lines for every bit of validation that goes on
in here as that would clutter the logs unnecessarily.

kurtismash commented:
Worth adding logs to say what folder/sourceId it's searching for as it does the recursive thing?

techncl commented on src/main/scala/uk/gov/nationalarchives/Lambda.scala:
```diff
@@ -103,12 +112,16 @@ class Lambda extends RequestStreamHandler {
         val message = generateTitleDescriptionSlackMessage(config.apiUrl, folderUpdateRequest)
         for {
+          _ <- log(s"Updating entity ${folderUpdateRequest.updateEntityRequest.ref}")
+          _ <- log(s"Sending\n$message\nto slack")
```
Shouldn't this come after `entitiesClient.updateEntity(folderUpdateRequest.updateEntityRequest)`, just in case updating the entity fails?

MancunianSam replied:
Good point, I've moved it.

techncl replied:
Oh no, I meant shouldn't the `log(s"Sending\n$message\nto slack")` message come after `entitiesClient.updateEntity(folderUpdateRequest.updateEntityRequest)` because how it is now, `entitiesClient.updateEntity(folderUpdateRequest.updateEntityRequest)` happens before the logging and  sending a message to slack is the first thing that is logged.

MancunianSam replied:
Ah I see what you mean. I've put "updating entity" back and moved "sending to slack" down a bit.

MancunianSam commented:
I've added in a line which shows which folder names/source ids are being searched for. I've added them to one log line as a comma separated list rather than a log line for each.

techncl commented on src/main/scala/uk/gov/nationalarchives/Lambda.scala:
```diff
@@ -59,6 +65,7 @@ class Lambda extends RequestStreamHandler {
       entitiesClient <- entitiesClientIO
       secretName = config.secretName
 
+      _ <- log(s"Searching for Source Ids ${folderRowsSortedByParentPath.map(_.name).mkString(",")}")
```
Could these names be sensitive?

MancunianSam replied:
I don't think so at this point for judgments and by the time we do get anything sensitive, we'll be encrypting the logs. At this point, it's just dumping stuff in there and seeing what's useful so we can always change our minds.

techncl reviewed (approved):
👍
